Core/Core/Files/Model/UploadManager.swift

```diff
@@ -36,6 +36,7 @@ open class UploadManager: NSObject, URLSessionDelegate, URLSessionTaskDelegate,
     var process: ProcessManager = ProcessInfo.processInfo
     var environment: AppEnvironment { .shared }
     private var validSession: URLSession?
+    private var task: APITask?
```

Comment:
The side effect of this change is that we only keep track of the last submission's API task. In case we have multiple submissions being uploaded the second one will overwrite this property so the task cancellation will break in case the first task's background process is terminated (it will cancel the second upload instead of the first one).

Reply:
Fixed!